.github/workflows/release.yaml

```diff
@@ -1,64 +1,80 @@
       - name: Set up Gren
-        if: github.event_name == 'repository_dispatch'
         run: npm install github-release-notes -g
       - name: Cache Dependencies
-        if: github.event_name != 'repository_dispatch'
         uses: actions/cache@v1
         with:
           path: ~/.m2/repository
           key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
           restore-keys: |
             ${{ runner.os }}-maven-
-      - name: Build Project
-        if: github.event_name != 'repository_dispatch'
-        run: mvn clean install
-      - name: Setup settings.xml
-        if: github.event_name == 'repository_dispatch'
+      - name: Code Checkout
         run: |
-          echo "<settings><servers><server><id>${{ secrets.OSSRH_ID }}</id><username>${{ secrets.OSSRH_USERNAME }}</username><password>${{ secrets.OSSRH_TOKEN }}</password></server></servers><profiles><profile><id>${{ secrets.OSSRH_ID }}</id><activation><activeByDefault>true</activeByDefault></activation><properties><gpg.executable>gpg</gpg.executable><gpg.passphrase>${{ secrets.GPG_PASSPHRASE}}</gpg.passphrase></properties></profile></profiles></settings>" > /home/runner/.m2/settings.xml
+          git init
+          git config --global user.name "github-actions"
+          git config --global user.email "github-actions@github.com"
+          git remote add origin "https://$GITHUB_ACTOR:${{ secrets.ACCESS_TOKEN }}@github.com/$GITHUB_REPOSITORY.git"
+          git fetch
+          git checkout master
+          git branch --set-upstream-to=origin/master
+          git pull
+          echo "#### Listing files after clone ####"
+          ls -a
+      - name: Update Release Version ${{steps.metadata.outputs.release-version}}
+        run: |
+          mvn versions:set -DnewVersion=${{steps.metadata.outputs.release-version}} -DgenerateBackupPoms=false -DprocessAllModules=true
+          git add -A
+          git commit -m"Automated version update: ${{steps.metadata.outputs.release-version}}"
+          git push
       - name: Publish to Maven Central
-        if: github.event_name == 'repository_dispatch'
         uses: samuelmeuli/action-maven-publish@v1
         with:
           gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
           gpg_passphrase: ${{ secrets.GPG_PASSPHRASE }}
           nexus_username: ${{ secrets.OSSRH_USERNAME }}
           nexus_password: ${{ secrets.OSSRH_TOKEN }}
           maven_goals_phases: clean deploy --batch-mode --settings /home/runner/.m2/settings.xml
```

Comment:
We should add `-DskipTests` I think, because we should build/test with maven before getting here.

Reply:
added  -DskipTests